src/flb_config.c

```diff
@@ -970,3 +980,55 @@ int flb_config_load_config_format(struct flb_config *config, struct flb_cf *cf)
 
     return 0;
 }
+
+int flb_config_task_map_resize(struct flb_config *config, size_t new_size)
+{
+    struct flb_task_map *new_task_map;
+
+    if (new_size == config->task_map_size) {
+        return 0;
+    }
+
+    if (new_size == 0) {
+        if (config->task_map != NULL) {
+            flb_free(config->task_map);
+
+            config->task_map = NULL;
+            config->task_map_size = 0;
+        }
+
+        return 0;
+    }
+
+    if (config->task_map == NULL) {
+        new_task_map = flb_calloc(new_size, sizeof(struct flb_task_map));
+    }
+    else {
+        new_task_map = flb_realloc(config->task_map, new_size * sizeof(struct flb_task_map));
+    }
+
+    if (new_task_map == NULL) {
+        return -1;
```

Comment:
missing flb_errno();

Reply:
fixed in commit 0c3b65628bdfe141fc1036babc86937be7c25c85